app/(main)/settings/components/github-connection-button.tsx

```diff
@@ -2,44 +2,37 @@
 
 import { Button } from "@/components/ui/button";
 import { cn } from "@/lib/utils";
+import { useRouter } from "next/navigation";
 import { useActionState } from "react";
 
 type ButtonWithActionProps = {
 	action: () => Promise<void>;
+	children: React.ReactNode;
+	className?: string;
 };
 
-const buttonClassNames = "w-fit font-avenir text-sm font-medium";
+export function GitHubConnectionButton({
+	action,
+	children,
+	className,
+}: ButtonWithActionProps) {
+	const buttonClassNames = "w-fit font-avenir text-sm font-medium";
+	const router = useRouter();
 
-export function GitHubConnectButton({ action }: ButtonWithActionProps) {
-	const [_, formAction, isPending] = useActionState(async () => {
-		await action();
-	}, null);
-	return (
-		<form>
-			<Button
-				className={buttonClassNames}
-				type="submit"
-				formAction={formAction}
-				disabled={isPending}
-			>
-				Connect
-			</Button>
-		</form>
-	);
-}
-export function GitHubDisconnectButton({ action }: ButtonWithActionProps) {
 	const [_, formAction, isPending] = useActionState(async () => {
 		await action();
+		router.refresh();
```

Comment:
Next.js has mechanism to execute `router.refresh` the client when the cache is revalidated in Route Handler or Server Action with `revalidatePath` or `revalidateTag`(undocumented).

I think it would be more generic if router.refresh is executed according to the action of the props, rather than router.refresh being executed whenever the action is executed, so I think it would be better to use revalidatePath and revalidateTag on the action side. I think it would be better to use revalidatePath and revalidateTag on the action side.

However, since this component has a well-defined use case and there are no major problems with it as it is now, I thought it was fine as it is in this pull request.